lib/index.ts

```diff
@@ -0,0 +1,261 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE
+
+const lighthouse = require('lighthouse');
+import {ChromeLauncher} from 'lighthouse/lighthouse-cli/chrome-launcher';
+const perfConfig: any = require('lighthouse/lighthouse-core/config/perf.json');
+const opn = require('opn');
+
+const Sheets = require('./sheets/index');
+const Chart = require('./chart/chart');
+const metrics = require('./metrics');
+const expectations = require('./expectations');
+const {upload} = require('./upload');
+const messages = require('./utils/messages');
+
+import {MainOptions, FeatureFlags, AuthorizeCredentials, LighthouseResults, MetricsResults, DriveResponse, TermWritableStream, PWMetricsResults, SheetsConfig, ExpectationMetrics} from '../types/types';
+
+const MAX_LIGHTHOUSE_TRIES = 2;
+const SIGINT = 'SIGINT';
+const SIGINT_EXIT_CODE = 130;
+const getTimelineViewerUrl = (id: string) => `https://chromedevtools.github.io/timeline-viewer/?loadTimelineFromURL=https://drive.google.com/file/d//${id}/view?usp=drivesdk`
+
+class PWMetrics {
+  flags: FeatureFlags = {
+    runs: 1,
+    submit: false,
+    upload: false,
+    view: false,
+    expectations: false,
+    output: false,
+    // @todo remove when new lighthouse version be released, because -https://github.com/GoogleChrome/lighthouse/pull/1778
+    disableCpuThrottling: false
+  };
+  runs: number;
+  sheets: SheetsConfig;
+  expectations: ExpectationMetrics;
+  clientSecret: AuthorizeCredentials;
+  tryLighthouseCounter: number;
+  launcher: ChromeLauncher | undefined = undefined;
+
+  constructor(public url: string, opts: MainOptions) {
+    this.flags = Object.assign({}, this.flags, opts.flags);
+    this.runs = this.flags.runs;
+    this.sheets = opts.sheets;
+    this.clientSecret = opts.clientSecret;
+    this.tryLighthouseCounter = 0;
+
+    if (this.flags.expectations) {
+      if (this.expectations) {
+        expectations.validateMetrics(this.expectations);
+        this.expectations = expectations.normalizeMetrics(this.expectations);
+      } else throw new Error(getMessageWithPrefix('ERROR', 'NO_EXPECTATIONS_FOUND'));
+    }
+  }
+
+  async start() {
+    const runs = Array.apply(null, {length: +this.runs}).map(Number.call, Number);
+    let metricsResults: MetricsResults[] = [];
+
+    // Kill spawned Chrome process in case of ctrl-C.
+    process.on(SIGINT, async() => {
+      if (this.launcher) {
+        try {
+          await this.killLauncher();
+          process.exit(SIGINT_EXIT_CODE);
+          console.log(messages.getMessage('CLOSING_CHROME'));
+        } catch (error) {
+          throw error;
```

Comment:
Way to catch errors for async/await is using try/catch
https://tc39.github.io/ecmascript-asyncawait/#examples